src/components/Todo/EditionForm.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import Datepicker from "react-datepicker";
+
+export default class EditionForm extends React.Component {
+  state = {
+    editTodoName: this.props.todo.name,
+    editDueDate: this.props.todo.dueDate
+   }
+
+  render() {
+
```

Comment:
If you can, try to start destructuring at the start of the render function.
```javascript
const { onUpdate, todo } = this.props;
const { editTodoName, editDueDate } = this.state;
const canSubmit = editTodoName === "" || editDueDate === null;

...

<form
    onSubmit={e => {
    e.preventDefault();
    if (!canSubmit) return;
    onUpdate(todo.id, { name: editTodoName, dueDate: editDueDate })
  }}
>

...

<button
  type="submit"
  className="button is-primary is-small"
  disabled={!canSubmit}
>
  Ok
</button>
```

Reply:
tu me dis "try" mais je copie/colle ton truc right ?

Reply:
pk on met pas "onCancel" dans la même const que "onUpdate" et "todo" du coup ?

Reply:
j'ai copié/collé ton truc and it is not working anymore mdr i am désespérée

Reply:
nvm it worked